docs/react-v9/contributing/rfcs/react-components/convergence/extended-design-tokens.md

```diff
@@ -0,0 +1,442 @@
+| ------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Option A: Update existing components with var() fallbacks** | • Existing consumers automatically get the FST system without additional work beyond updating their packages<br>• Everyone stays in sync and should have equal capability improving interop and portability | • Once the changes are made, we are somewhat stuck with them (until the next major version)<br>• All consumers will pull these changes which means some increases in bundle size (gzipped impact seems minor based on our test PR so far).                                                                                                                                                                                                                                                               |
+| Option B: Custom style hooks                                  | • Fully opt in<br>• Removes component changes at the root and could allow us to ship separately from the core packages meaning breaking changes could be managed more easily in a separate package          | • performance regressions were noted in testing. This might be due to the fact that styles were generated and inserted in runtime. Additionally, it depends on React's context which re-renders components.<br>• Because it's opt in we wouldn't have uniform adoption<br><br>Opportunity: If we can fix the performance issues or understand them better this might be a really interesting option with lower risk. Can we investigate using these styles hooks but with Griffel outputting static CSS? |
```

Comment:
I'd test both AOT and extracted CSS.

https://griffel.js.org/react/ahead-of-time-compilation/introduction

Reply:
How do you envision these custom hooks being implemented? One of the limitations of the existing custom style hooks is there no "[inheritance](https://github.com/microsoft/fluentui/issues/30308)" and you can only supply a single custom function. This means if we use the custom style hooks in Fluent, Fluent's users cannot also use them.

Reply:
> How do you envision these custom hooks being implemented? One of the limitations of the existing custom style hooks is there no "[inheritance](https://github.com/microsoft/fluentui/issues/30308)" and you can only supply a single custom function. This means if we use the custom style hooks in Fluent, Fluent's users cannot also use them.

Ah I didn't actually realize that limitation. That's a really good point, so this would only function if users kept default styling.